lib/taxonomy/world_taxonomy.rb

```diff
@@ -0,0 +1,20 @@
+module Taxonomy
+  class WorldTaxonomy
+    def initialize(adapter: RedisCacheAdapter.new, tree_builder_class: Tree)
+      @adapter = adapter
+      @tree_builder_class = tree_builder_class
+    end
+
+    def all_world_taxons
+      @_world_taxon_list ||= world_taxon_branches
+    end
+
+  private
+
+    def world_taxon_branches
+      @_branches ||= @adapter.world_taxon_data.map do |world_taxon_hash|
+        @tree_builder_class.new(world_taxon_hash).root_taxon
+      end
```

Comment:
This is a bit odd, as the concept of "branches" is a bit confused here.

With the Topic Taxonomy, branches mostly refer to level one taxons. As far as I can work out here, the code relies on the structure being the same as the Topic Taxonomy, treating the World taxon like it's a level one taxon, whereas really it's the root of the Worldwide Taxonomy (effectively level 0).

I don't think this is worth trying to unpick further now, but lets put something in the backlog.